File: check_frontmatter.js
const fs = require('fs');
const path = require('path');

const rootDir = 'c:/Users/lordh/coding/hackathon1_book/docs';

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);
        if (stat.isDirectory()) {
            walkDir(filePath);
        } else if (file.endsWith('.md')) {
            try {
                const content = fs.readFileSync(filePath, 'utf8');
                const firstLine = content.trim().split('\n')[0];
                if (!firstLine.trim().startsWith('---')) {
                    console.log(`MISSING FRONTMATTER: ${filePath}`);
                }
            } catch (err) {
                console.error(`Error reading ${filePath}: ${err}`);
            }
        }
    }
}

console.log("Scanning for files missing frontmatter...");
walkDir(rootDir);
console.log("Scan complete.");

File: check_frontmatter.py
import os

root_dir = 'c:/Users/lordh/coding/hackathon1_book/docs'

print("Scanning for files missing frontmatter...")
for dirpath, dirnames, filenames in os.walk(root_dir):
    for filename in filenames:
        if filename.endswith('.md'):
            filepath = os.path.join(dirpath, filename)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    first_line = f.readline().strip()
                    if first_line != '---':
                        print(f"MISSING FRONTMATTER: {filepath}")
            except Exception as e:
                print(f"Error reading {filepath}: {e}")
print("Scan complete.")
